README.md

```diff
@@ -3,4 +3,356 @@
+
+If you haven’t already installed the Salesforce CLI, follow 
+[these steps](https://developer.salesforce.com/docs/atlas.en-us.sfdx_setup.meta/sfdx_setup/sfdx_setup_install_cli.htm). 
+
+If you already have the Salesforce CLI installed, make sure it is updated to the latest release candidate, which contains the Python 
+functions support:
+
+```sh
+sfdx update stable-rc
+```
+
+This will update both the `sfdx` and `sf` commands.  The installed version of `sf` should be `1.59.0` or higher.
+
+You can check this with `sf --version`
+
+### [Create a SFDX Project](https://developer.salesforce.com/docs/platform/functions/guide/create-dx-project.html)
+
+Functions must be located within a SFDX project, so let’s create one.  
+
+```sh
+sf generate project --name PythonFunctionsAlpha
+```
+
+Some of the following commands need to be run from within the SFDX project directory so change into that directory now with
+
+```sh
+cd PythonFunctionsAlpha
+```
+
+You should also edit the `config/project-scratch-def.json` file to include the **`Functions`** feature.  After modification, your 
+file should look similar to the following:
+
+```json
+{
+  "orgName": "SomeOrgName",
+  "edition": "Developer",
+  "features": ["EnableSetPasswordInApi", "Functions"],
+  "settings": {
+    "lightningExperienceSettings": {
+      "enableS1DesktopEnabled": true
+    },
+    "mobileSettings": {
+      "enableS1EncryptedStoragePref2": false
+    }
+  }
+}
+```
+
+The above is needed for connecting and deploying Functions in scratch orgs which is 
+[the recommended workflow](https://developer.salesforce.com/docs/platform/functions/guide/connect-dev-org.html) when working 
+with Functions.
+
+And, to deploy any Salesforce Function, your SFDX project needs to be a git repo.  This is because the deployment process 
+uses git tracked changes to figure out what to deploy.  Run the following commands to setup git:
+
+```sh
+git init
+```
+
+> It is not a requirement to push your code to GitHub or any other code hosting site.  Committing locally will work fine for deploying.  
+
+### Connect Your Org
+
+You’ll need to configure your Salesforce org to develop and invoke Salesforce Functions. Develop your functions in scratch 
+orgs with Dev Hub or in sandbox orgs.  Follow the steps on 
+[this page](https://developer.salesforce.com/docs/platform/functions/guide/configure_your_org.html) to ensure everything 
+is setup correctly first.
+
+Once your Org is configured you can login and set it as the default Dev Hub with the following command:
+
+```sh
+sf login org --alias PythonOrg --set-default-dev-hub --set-default
+```
+
+This will make **PythonOrg** the default Dev Hub for subsequent commands.
+
+Then create a scratch org:
+
+```sh
+sfdx force:org:create \
+  --definitionfile config/project-scratch-def.json \
+  --setalias PythonScratch \
+  --setdefaultusername
```

Comment:
When I ran this command, I encountered an error:

```
Functions is not a valid Features value
```

This was when I'd logged into my dev org in the prior step, which it turns out did not have the Functions licence enabled.

Should we mention the need to have a functions licence enabled in the parent org first?